tests/tod/test_tod_agents_and_teachers.py

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Tests different (more complicated) slot metrics.
+"""
+
+import unittest
+
+import copy
+import parlai.core.tod.tod_core as tod_core
+import parlai.core.tod.tod_test_utils.test_agents as test_agents
+
+
+class TestTodAgentsAndTeachersBase(unittest.TestCase):
```

Comment:
Hmm these classes might be especially good to add docstrings for, because if these tests break it'd be useful to know what they do in order to help debug them

Reply:
...good call, thanks!